File: app/src/main/java/com/vinodhn/android/backtap/SettingsActivity.java
package com.vinodhn.android.backtap;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.Toast;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.preference.ListPreference;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;
import androidx.preference.PreferenceManager;

public class SettingsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.settings_activity);
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.settings, new SettingsActivity.SettingsFragment())
                .commit();
        ActionBar actionBar = getSupportActionBar();
        if(actionBar != null){
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
        actionBar.setTitle("BackTap Action Settings");
    }

    public static class SettingsFragment extends PreferenceFragmentCompat {
        @Override
        public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
            setPreferencesFromResource(R.xml.accessibility_preferences, rootKey);

            final ListPreference mDoubleTapActionPref = findPreference("double");
            final ListPreference mTripleTapActionPref = findPreference("triple");
            final Preference mAboutPreference = findPreference("About");
            final Preference mDonatePreference = findPreference("Donate");
            SharedPreferences mSharedPrefs = PreferenceManager.getDefaultSharedPreferences(getActivity());
            final SharedPreferences.Editor mEditor = mSharedPrefs.edit();

            if(mDoubleTapActionPref != null){
                mDoubleTapActionPref.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
                    @Override
                    public boolean onPreferenceChange(Preference preference, Object newValue) {
                        mEditor.putInt(getString(R.string.double_tap_action_id), mDoubleTapActionPref.findIndexOfValue(newValue.toString()));
                        mEditor.commit();
                        return true;
                    }
                });
            }

            if(mTripleTapActionPref != null){
                mTripleTapActionPref.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
                    @Override
                    public boolean onPreferenceChange(Preference preference, Object newValue) {
                        mEditor.putInt(getString(R.string.triple_tap_action_id), mTripleTapActionPref.findIndexOfValue(newValue.toString()));
                        mEditor.commit();
                        return true;
                    }
                });
            }
            if(mAboutPreference != null){
                mAboutPreference.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
                    @Override
                    public boolean onPreferenceClick(Preference preference) {
                        AlertDialog.Builder mBuilder = new AlertDialog.Builder(getActivity());
                        mBuilder.setMessage("A clone of the accessibility feature in iOS 14 and Android 11.\nCreated by Vinodh N.\nVersion 1.0 Alpha").setTitle("About BackTap");
                        AlertDialog mDialog = mBuilder.create();
                        mDialog.show();
                        return false;
                    }
                });
            }

            if(mDonatePreference != null){
                mDonatePreference.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
                    @Override
                    public boolean onPreferenceClick(Preference preference) {
                        Toast.makeText(getContext(), "Donate coming soon.", Toast.LENGTH_LONG).show();
                        return false;
                    }
                });
            }

        }
    }

}

File: app/src/main/java/com/vinodhn/android/backtap/SettingsActivity.kt
package com.vinodhn.android.backtap

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.provider.Settings
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.ListPreference
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import androidx.preference.PreferenceManager

class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.settings_activity)
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.settings, SettingsFragment())
            .commit()
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setTitle("BackTap Action Settings")
    }

    class SettingsFragment : PreferenceFragmentCompat() {
        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
            setPreferencesFromResource(R.xml.accessibility_preferences, rootKey)
            val mDoubleTapActionPreference = findPreference<ListPreference>("double")
            val mTripleTapActionPreference = findPreference<ListPreference>("triple")
            val mAboutPreference = findPreference<Preference>("About")
            val mDonatePreference = findPreference<Preference>("Donate")
            val mSharedPreference: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(activity)
            val mEditor: SharedPreferences.Editor = mSharedPreference.edit()

            mDoubleTapActionPreference?.onPreferenceChangeListener = Preference.OnPreferenceChangeListener { preference, newValue ->

                if(mDoubleTapActionPreference != null) {
                    mEditor.putInt(getString(R.string.double_tap_action_id), mDoubleTapActionPreference.findIndexOfValue(newValue.toString()))
                    mEditor.commit()
                }

                true
            }

            mTripleTapActionPreference?.onPreferenceChangeListener = Preference.OnPreferenceChangeListener { preference, newValue ->

                if(mTripleTapActionPreference != null){
                    mEditor.putInt(getString(R.string.triple_tap_action_id), mTripleTapActionPreference.findIndexOfValue(newValue.toString()))
                    mEditor.commit()
                }

                true
            }

            mAboutPreference?.setOnPreferenceClickListener { preference ->

                val mBuilder: AlertDialog.Builder? = activity?.let { AlertDialog.Builder(it) }
                mBuilder?.setMessage("A clone of the accessibility feature in iOS 14 and Android 11.\nCreated by Vinodh N.\nVersion 1.0 Alpha")
                mBuilder?.setTitle("About BackTap")
                val mDialog: AlertDialog? = mBuilder?.create()
                mDialog?.show()

                false
            }

            mDonatePreference?.setOnPreferenceClickListener { preference ->
                Toast.makeText(context, "Donate coming soon", Toast.LENGTH_LONG).show()
                false
            }

        }
    }
}